src/main/java/com/gooddata/md/MetadataService.java

```diff
@@ -97,6 +101,31 @@ public MetadataService(RestTemplate restTemplate) {
     }
 
     /**
+     * Retrieves a collection of objects corresponding to the supplied collection of URIs.
+     *
+     * @param project project that contains the objects to be retrieved
+     * @param uris collection of URIs
+     * @return collection of metadata objects corresponding to the supplied URIs
+     */
+    public Collection<Obj> getObjsByUris(Project project, Collection<String> uris) {
+        notNull(project, "project");
+        notNull(project.getId(), "project.id");
+        notNull(uris, "uris");
+
+        try {
+            final BulkGet result = restTemplate.postForObject(BulkGet.URI, new BulkGetUris(uris), BulkGet.class, project.getId());
+
+            if (result != null) {
+                return result.getItems();
+            } else {
+                throw new GoodDataException("empty response from API call");
+            }
+        } catch (RestClientException e) {
+            throw new GoodDataException("Unable to get objects. Some of the supplied URIs may be malformed.");
```

Comment:
I suggest to include the `RestClientException` as the cause of the `GoodDataException`